lib/elixir/lib/map.ex

```diff
@@ -565,9 +568,11 @@ defmodule Map do
   """
   @spec pop(map, key, value) :: {value, map}
   def pop(map, key, default \\ nil) do
-    case map do
-      %{^key => value} -> {value, delete(map, key)}
-      %{} -> {default, map}
+    case fetch(map, key) do
+      {:ok, value} ->
+        {value, delete(map, key)}
+      :error ->
+        {default, map}
```

Comment:
It was recently optimized like that for `Access.get` in https://github.com/elixir-lang/elixir/commit/2a900db540697504758da11284125a14591497fe

So it seems pretty wired to me to have `Access` better optimized than `Map`.